election-day/obetsun/election_day.go

```diff
@@ -0,0 +1,38 @@
+package electionday
+
+import "strconv"
+
+// NewVoteCounter returns a new vote counter with
+// a given number of inital votes.
+func NewVoteCounter(initialVotes int) *int {
+	var counter int
+    counter = initialVotes
+    return &counter
+}
+
+// VoteCount extracts the number of votes from a counter.
+func VoteCount(counter *int) int {
+	if counter == nil {return 0}
+    return *counter
+}
+
+// IncrementVoteCount increments the value in a vote counter
+func IncrementVoteCount(counter *int, increment int) {
+	*counter+=increment
+}
+
+// NewElectionResult creates a new election result
+func NewElectionResult(candidateName string, votes int) *ElectionResult {
+	electionResult := ElectionResult{candidateName, votes}
+    return &electionResult
+}
+
+// DisplayResult creates a message with the result to be displayed
+func DisplayResult(result *ElectionResult) string {
+	return result.Name+" ("+strconv.Itoa(result.Votes)+")"
```

Comment:
ah, and here you can use fmt.Sprintf to construct the string, it's more elegant, `fmt.Sprintf("%s (%d)", result.Name, result.Votes)`.

Reply:
You're right, actually this exercise I did long ago, don't know why I did it this way.